clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -155,7 +155,16 @@ public void remove(String id) {
     public ConnectionState connectionState(String id) {
         return nodeState(id).state;
     }
-    
+
+    /**
+     * Check if we have attempted to connect to this node.
+     * @param id node id
+     * @return true if we have attempted to connect to the node
+     */
+    public boolean connectionAttempted(String id) {
```

Comment:
This method is a little weird. In `isConnected()` and `isReady()` we explicitly verifies `state != null`. Maybe it is more consistent to just add a `isDisconnected()` method in the `ClusterConnectionState` and let `NetworkClient.connectionFailed()` just call that method.